src/wallet/rpcwallet.cpp

```diff
@@ -1622,9 +1620,9 @@ static UniValue listsinceblock(const JSONRPCRequest& request)
     // when a reorg'd block is requested, we also list any relevant transactions
     // in the blocks of the chain that was detached
     UniValue removed(UniValue::VARR);
-    while (include_removed && paltindex && paltindex != pindex) {
+    while (include_removed && altheight && *altheight > height.value_or(-1)) {
```

Comment:
in commit 22abf4a61a Add findFork and findBlock to the Chain interface

Assuming that height can not be set makes no sense, since you moved the FindFork call to before the `!height` check. I'd prefer to write `*height` here.